src/contexts/GUIContextProvider.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import { createMuiTheme, ThemeProvider } from '@material-ui/core';
+import React from 'react';
+import useLocalStorage from '../hooks/useLocalStorage';
+import { light } from '../components';
+
+interface DrawerContextProps {
+  drawer: string;
+  setDrawer: React.Dispatch<React.SetStateAction<string>>;
+}
+interface GUIContextProviderProps {
+  children: React.ReactElement;
+}
+
+export const DrawerContext = React.createContext({} as DrawerContextProps);
+
+export function GUIContextProvider({ children }: GUIContextProviderProps): React.ReactElement {
+  const [drawer, setDrawer] = useLocalStorage('storageDrawer');
+  const value = { drawer, setDrawer };
+
+  return (
+    <ThemeProvider theme={createMuiTheme(light)}>
+      <DrawerContext.Provider value={value}>{children}</DrawerContext.Provider>
```

Comment:
This is not bad , but i like the pattern applied on other contexts where i created two contexts in the provider file: one context for values and the other for updating, so you separate concerns on your state ( values & updating )

It would be something like this:

```
    <DrawerContext.Provider value={{drawer}}>
      <UpdateDrawerContext.Provider value={{ setDrawer }}>
         {children}
      </UpdateDrawerContext.Provider>
    </DrawerContext.Provider>
``` 

Also i know that it was highly criticized, but managing the states with reducers makes the context extremely powerful  and simple for the children components using and updating this state. On this last one, is not a hard requirement since the usage is pretty basic.

Reply:
I see your point and your contexts too. I'd use Context-UpdateContext when reducers are sensible to use at all in this context, if that's ok

Reply:
> tbh I'm thinking now to get rid of that context and not to merge this at all, as the delay that it takes for drawer to open is just annoyingly long

Im not sure I agree - the open/close of the drawer is pretty fine to my eyes - I dont see any delay or glitch

Reply:
I don't think it's to do with animation, but I'll look into the transitions one more time. Or maybe it's my computer was boiled by the end of the day :)